src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -2028,4 +2028,100 @@ BOOST_AUTO_TEST_CASE(GradientReparametrizationLayerBetaTest)
   BOOST_REQUIRE_LE(CheckGradient(function), 1e-4);
 }
 
+/**
+ * Simple sequential module residual block test.
+ */
+BOOST_AUTO_TEST_CASE(SimpleSequentialLayerResidualBlockTest)
+{
+  arma::mat outputA /* sequential */, outputB /* residual */, input,
+      deltaA /* sequential */, deltaB /* residual */;
+  Sequential<>* sequential = new Sequential<>(true, false);
+  Sequential<>* residual = new Sequential<>(true, true);
+
+  Linear<>* linearA = new Linear<>(10, 10);
+  linearA->Parameters().randu();
+  linearA->Reset();
+  Linear<>* linearB = new Linear<>(10, 10);
+  linearB->Parameters().randu();
+  linearB->Reset();
+
+  // Add the same layers(with the same parameters) to both normal Sequential and
+  // residual Sequential object.
+  sequential->Add(linearA);
+  sequential->Add(linearB);
+
+  residual->Add(linearA);
+  residual->Add(linearB);
+
+  // Test the Forward function(Pass the same input to both).
+  input = arma::randu(10, 1);
+  sequential->Forward(std::move(input), std::move(outputA));
+  residual->Forward(std::move(input), std::move(outputB));
+
+  CheckMatrices(outputA, outputB - input);
+
+  // Test the Backward function(Pass the same error to both).
+  sequential->Backward(std::move(input), std::move(input), std::move(deltaA));
+  residual->Backward(std::move(input), std::move(input), std::move(deltaB));
+
+  CheckMatrices(deltaA, deltaB - input);
+
+  delete sequential;
+  delete residual;
+  delete linearA;
+  delete linearB;
+}
+
+/**
+ * Sequential layer numerical gradient test.
+ */
+BOOST_AUTO_TEST_CASE(GradientSequentialLayerTest)
+{
+  // Linear function gradient instantiation.
+  struct GradientFunction
+  {
+    GradientFunction()
+    {
+      input = arma::randu(10, 1);
+      target = arma::mat("1");
+
+      model = new FFN<NegativeLogLikelihood<>, NguyenWidrowInitialization>();
```

Comment:
Do you think we could add another ``Sequential`` layer test with ``residual = true``?